### PR TITLE
Ops 89 distributed monitoring

### PR DESCRIPTION
**Pull request by jameswarreninss**

### CODE REVIEW

File: setup/alerting/icinga2/tasks/configure_zones.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+- name: create zones directory structure on master
+  file:
+    path: "/etc/icinga2/zones.d/{{ item }}"
+    state: directory
+    owner: root
```
Comment (inss-mark) [inline setup/alerting/icinga2/tasks/configure_zones.yml]: Should this be icinga:icinga and 750?
Also watch out as ansible expects 0750

File: setup/alerting/icinga2/tasks/configure_zones.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    - { file: "users.conf", directory: "zones.d/global-templates/" }
+  notify: restart icinga2
+
+- name: template dynamic configuration
+  template:
+    src: "{{ item.file }}.j2"
+    dest: "/etc/icinga2/zones.d/{{ item.directory }}/{{ item.file }}"
+    owner: root
+    group: root
+    mode: 0644
```
Comment (inss-mark) [inline setup/alerting/icinga2/tasks/configure_zones.yml]: World readable?

File: setup/alerting/icinga2/tasks/configure_zones.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    - { file: "downtimes.conf", directory: "global-templates" }
+  notify: restart icinga2
+
+- name: template icinga2 admin contacts
+  template:
+    src: contact.conf.j2
+    dest: "/etc/icinga2/zones.d/{{ icinga_zone }}/contact_{{ item.key }}.conf"
+    owner: root
+    group: root
+    mode: 0644
```
Comment (inss-mark) [inline setup/alerting/icinga2/tasks/configure_zones.yml]: World readable

File: setup/alerting/icinga2/tasks/database.yml [hunk 4]
```diff
@@ -1,17 +1,21 @@
 ---
-- name: Set debconf selections to install postgres db configuration for icinga2
+- name: set debconf selections to install postgres db configuration for icinga2
   debconf:
     name: icinga2-ido-pgsql
     question: "icinga2-ido-pgsql/dbconfig-install"
     value: true
     vtype: boolean
+  register: deb_conf_icinga_pgsql_db
+  changed_when: deb_conf_icinga_pgsql_db.current|urlencode|lower != deb_conf_icinga_pgsql_db.previous|urlencode|lower
```
Comment (inss-mark) [inline setup/alerting/icinga2/tasks/database.yml]: Maybe comment for your future or predecessor's health ;)

File: setup/alerting/icinga2/tasks/database.yml [hunk 5]
```diff
@@ -35,19 +41,27 @@
...
+      notify: restart icinga2
+      when: icinga2_current_apppass.stdout[2:] != icinga_database_setup_icinga2_password
+  no_log: true
 
 - name: reset icinga2-ido-pgsql database password
   become_user: postgres
   postgresql_user:
     name: icinga2
-    password: "{{ icinga.icinga_database_setup.icinga2.password }}"
+    password: "{{ icinga_database_setup_icinga2_password }}"
```
Comment (inss-mark) [inline setup/alerting/icinga2/tasks/database.yml]: :+1:

File: setup/alerting/icinga2/tasks/generate_tickets.yml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  shell: "icinga2 pki ticket --cn {{ this_satellite }}.insolvency-development.co.uk"
+  register: satellite_key
+
+- name: create file to show ticket already generated for that node
+  template:
+    src: "icinga_key.j2"
+    dest: "/etc/icinga2/{{ this_satellite | lower }}_ticket_generated"
+    owner: root
+    group: root
+    mode: 0644
```
Comment (inss-mark) [inline setup/alerting/icinga2/tasks/generate_tickets.yml]: Key is going to be world readable too :(

File: setup/alerting/icinga2/templates/constants.conf.j2 [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+/* The directory which you use to store additional plugins which ITL provides user contributed command definitions for.
+ * Check the documentation, chapter "Plugins Contribution", for details.
+ */
+const PluginContribDir = "/usr/lib/nagios/plugins"
+
+/* Our local instance name. By default this is the server's hostname as returned by `hostname --fqdn`.
+ * This should be the common name from the API certificate.
+ */
+const NodeName = "LCDEVFINSMGMT01.insolvency-development.co.uk"
+
```
Comment (inss-mark) [inline setup/alerting/icinga2/templates/constants.conf.j2]: Hardcoded nodename!

File: setup/alerting/icingaweb2/tasks/icingaweb2_configure.yml [hunk 8]
```diff
@@ -13,28 +13,15 @@
   postgresql_db:
     name: icingaweb2
     owner: icingaweb2
+  register: icinga_create_db_state
 
 - name: load the icingaweb2 database model
-  raw: export PGPASSWORD={{ icinga.icinga_database_setup.icingaweb2.password }} && psql -U icingaweb2 -d icingaweb2 < /usr/share/icingaweb2/etc/schema/pgsql.schema.sql
+  raw: export PGPASSWORD={{ icinga_database_setup_icingaweb2_password }} && psql -U icingaweb2 -d icingaweb2 < /usr/share/icingaweb2/etc/schema/pgsql.schema.sql
+  changed_when: icinga_create_db_state.changed == true
```
Comment (inss-mark) [inline setup/alerting/icingaweb2/tasks/icingaweb2_configure.yml]: Ticket this up for fixing when ansible 2.8 is deployed.

File: setup/alerting/icingaweb2/tasks/icingaweb2_create_user.yml [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+- name: create new user if it doesn't exist
+  block:
+  - name: template sql user scripts
+    template:
+      src: new_user.sql.j2
+      dest: "/tmp/{{ item.key }}.sql"
+    no_log: true
+  - name: run sql user scripts into the database
+    raw: export PGPASSWORD={{ icinga_database_setup_icingaweb2_password }} && psql -U icingaweb2 -d icingaweb2 < /tmp/{{ item.key }}.sql
+    no_log: true
```
Comment (inss-mark) [inline setup/alerting/icingaweb2/tasks/icingaweb2_create_user.yml]: would be more secure to keep it out of environment vars and use inline with psql

File: setup/alerting/icingaweb2/tasks/icingaweb2_install.yml [hunk 10]
```diff
@@ -7,7 +7,7 @@
 - name: Set permissions on PHP Unix Socket
   lineinfile:
     path: /etc/php/7.0/fpm/pool.d/www.conf
-    regexp: '^;listen.mode = 0660'
+    regexp: '.*listen.mode.*'
     line: 'listen.mode = 6666'
```
Comment (inss-mark) [inline setup/alerting/icingaweb2/tasks/icingaweb2_install.yml]: listen.mode of 6666 might wanna secure that

File: setup/alerting/icingaweb2/templates/resources.ini.j2 [hunk 11]
```diff
@@ -5,7 +5,7 @@ host = "localhost"
 port = "5432"
 dbname = "icingaweb2"
 username = "icingaweb2"
-password = "{{ icinga.icinga_database_setup.icingaweb2.password }}"
+password = "{{ icinga_database_setup_icingaweb2_password }}"
 charset = ""
 persistent = "0"
```
Comment (inss-mark) [inline setup/alerting/icingaweb2/templates/resources.ini.j2]: Should persistent be 0? Surely a persistent DB connection would be better.  Answers on a postcard?

File: setup/alerting/linux_client/tasks/main.yml [hunk 12]
```diff
@@ -41,4 +41,4 @@
 - name: Allow external access for monitoring agent
   ufw:
     rule: allow
-    port: "{{ alerting.agent_port }}"
+    port: "{{ alerting.agent_port }}"
```
Comment (inss-mark) [inline setup/alerting/linux_client/tasks/main.yml]: Should this be alerting_agent_port?

File: setup/alerting/linux_client/templates/nrpe.cfg [hunk 13]
```diff
@@ -73,7 +73,8 @@ nrpe_group=nagios
 #
 # NOTE: This option is ignored if NRPE is running under either inetd or xinetd
 
-allowed_hosts=127.0.0.1,{{ hostvars[groups['management'][0]]['ansible_host'] }}{% if env == "LCDEV" %},{{ ansible_host[:-2] }}1 {% endif %}
+# For localdev we need to allow the vagrant gateway ip also.
+allowed_hosts=127.0.0.1,{{ hostvars[groups['management'][0]]['ansible_host'] }}{% if env == "LCDEV" %},{% for i in ansible_host.split('.') %}{% if loop.index < 4 %}{{ i }}.{% endif %}{% endfor %}1{% endif %}
```
Comment (inss-mark) [inline setup/alerting/linux_client/templates/nrpe.cfg]: Can haz comment ^ :)

File: setup/alerting/linux_client/templates/nrpe.cfg [hunk 14]
```diff
@@ -199,21 +200,21 @@ command[check_zombie_procs]=/usr/lib/nagios/plugins/check_procs -w 5 -c 10 -s Z
...
 command[check_rabbitmq_objects]=/usr/lib/nagios/plugins/check_rabbitmq_objects -H {{ ansible_hostname }} -u monitor -p {{ rabbitmq.monitor_password }}
 command[check_rabbitmq_overview]=/usr/lib/nagios/plugins/check_rabbitmq_overview -H {{ ansible_hostname }} -u monitor -p {{ rabbitmq.monitor_password }}
 command[check_rabbitmq_partition]=/usr/lib/nagios/plugins/check_rabbitmq_partition -H {{ ansible_hostname }} -u monitor -p {{ rabbitmq.monitor_password }}
 command[check_rabbitmq_queue]=/usr/lib/nagios/plugins/check_rabbitmq_queue -H {{ ansible_hostname }} -u monitor -p {{ rabbitmq.monitor_password }}
 command[check_rabbitmq_server]=/usr/lib/nagios/plugins/check_rabbitmq_server -H {{ ansible_hostname }} -u monitor -p {{ rabbitmq.monitor_password }}
-command[check_rabbitmq_watermark]=/usr/lib/nagios/plugins/check_rabbitmq_watermark -H {{ ansible_hostname }} -u monitor -p {{ rabbitmq.monitor_password }}
+command[check_rabbitmq_watermark]=/usr/lib/nagios/plugins/check_rabbitmq_watermark -H {{ ansible_hostname }} -u monitor -p {{ rabbitmq.monitor_password }}{% endif %}
```
Comment (inss-mark) [inline setup/alerting/linux_client/templates/nrpe.cfg]: Probably enough lines for rabbit and friends to be put in a templated/included file ;)

File: setup/alerting/linux_client/tasks/main.yml (hunk 12 above)
Comment (jameswarreninss) [reply setup/alerting/linux_client/tasks/main.yml]: Haven't updated the alerting linux modules so this is correct for now.

**Pull request merged**